spark/v3.4/spark-runtime/src/integration/java/org/apache/iceberg/spark/SmokeTest.java

```diff
@@ -117,11 +117,7 @@ public void testAlterTable() throws NoSuchTableException {
     sql("ALTER TABLE %s DROP PARTITION FIELD shard", tableName);
 
     table = getTable();
-    Assert.assertEquals("Table should have 4 partition fields", 4, table.spec().fields().size());
-    // VoidTransform is package private so we can't reach it here, just checking name
-    Assert.assertTrue(
-        "All transforms should be void",
-        table.spec().fields().stream().allMatch(pf -> pf.transform().toString().equals("void")));
+    Assert.assertTrue("Table should be unpartitioned", table.spec().isUnpartitioned());
```

Comment:
minor: I think it would be good to switch assertions to AssertJ in such a case - the project is trying to move away from Junit4-style assertions, and it would be great to have new assertions written using AssertJ to make the migration process easier

Reply:
Yeah, I've used the new assertion style whenever adding new checks or when modifying classes that migrated. Here, I modify the existing check so I followed the rest of the class. I was not sure it is worth to just switch a single check.